src/App/Pages/Accounts/LoginSsoPage.xaml.cs

```diff
@@ -0,0 +1,114 @@
+﻿using Bit.App.Models;
+using Bit.Core.Abstractions;
+using Bit.Core.Utilities;
+using System;
+using System.Threading.Tasks;
+using Bit.App.Utilities;
+using Xamarin.Forms;
+
+namespace Bit.App.Pages
+{
+    public partial class LoginSsoPage : BaseContentPage
+    {
+        private readonly IStorageService _storageService;
+        private readonly IMessagingService _messagingService;
+        private readonly IVaultTimeoutService _vaultTimeoutService;
+        private readonly LoginSsoPageViewModel _vm;
+        private readonly AppOptions _appOptions;
+
+        private AppOptions _appOptionsCopy;
+
+        public LoginSsoPage(AppOptions appOptions = null)
+        {
+            _storageService = ServiceContainer.Resolve<IStorageService>("storageService");
+            _messagingService = ServiceContainer.Resolve<IMessagingService>("messagingService");
+            _vaultTimeoutService = ServiceContainer.Resolve<IVaultTimeoutService>("vaultTimeoutService");
+            _messagingService.Send("showStatusBar", true);
+            _appOptions = appOptions;
+            InitializeComponent();
+            _vm = BindingContext as LoginSsoPageViewModel;
+            _vm.Page = this;
+            _vm.StartTwoFactorAction = () => Device.BeginInvokeOnMainThread(async () => await StartTwoFactorAsync());
+            _vm.StartSetPasswordAction = () =>
+                Device.BeginInvokeOnMainThread(async () => await StartSetPasswordAsync());
+            _vm.SsoAuthSuccessAction = () => Device.BeginInvokeOnMainThread(async () => await SsoAuthSuccessAsync());
+            _vm.CloseAction = async () =>
+            {
+                _messagingService.Send("showStatusBar", false);
+                await Navigation.PopModalAsync();
+            };
+            if (Device.RuntimePlatform == Device.Android)
+            {
+                ToolbarItems.RemoveAt(0);
+            }
+        }
+
+        protected override async void OnAppearing()
+        {
+            base.OnAppearing();
+            await _vm.InitAsync();
+            if (string.IsNullOrWhiteSpace(_vm.OrgIdentifier))
+            {
+                RequestFocus(_orgIdentifier);
+            }
+        }
+
+        private void CopyAppOptions()
+        {
+            if (_appOptions != null)
+            {
+                // create an object copy of _appOptions to persist values when app is exited during web auth flow
+                _appOptionsCopy = new AppOptions();
```

Comment:
Further clarification on this:  Turns out the app is clearing the `Uri` from `Options` when it gets sent to the background during the web auth flow (see `App.xaml.cs > SetTabsPageFromAutofill(...)`).  To avoid messing with that method, I'm creating a copy of `AppOptions` then restoring the values back to the original upon web auth completion. I'm using a helper method instead of a copy constructor so the original object can be repopulated, and I added the method to the `AppOptions` class so it won't be overlooked if/when future options are added.